ComponentKit/Core/CKComponentViewAttribute.h

```diff
@@ -56,14 +69,41 @@ struct CKComponentViewAttribute {
    |        |          | Updater is called if the attribute was previously applied and the value changes.              |
    |--------|----------|-----------------------------------------------------------------------------------------------|
    */
+  template<typename T, typename U, typename = typename std::enable_if<std::is_convertible<T, id>::value>::type>
   CKComponentViewAttribute(const std::string &ident,
-                           void (^app)(id view, id value),
-                           void (^unapp)(id view, id value) = nil,
-                           void (^upd)(id view, id oldValue, id newValue) = nil) :
+                           void (^app)(T view, U value),
+                           void (^unapp)(T view, U value),
+                           void (^upd)(T view, U oldValue, U newValue)) :
   identifier(ident),
-  applicator(app),
-  unapplicator(unapp),
-  updater(upd) {};
+  applicator([=](T view, U value) { app(view, value); }),
+  unapplicator(unapp ? ApplicatorFunc([=](T view, U value) { unapp(view, value); }) : ApplicatorFunc(nullptr)),
```

Comment:
Why wrap these in a C++ lambda? You're using std::function right? std::function accepts blocks directly.